src/Compilers/CSharp/Test/Semantic/Semantics/ConditionalOperatorTests.cs

```diff
@@ -1499,5 +1499,24 @@ public TestClass Self()
 
             CompileAndVerify(compilation, expectedOutput: "---");
         }
+
+        [Fact, WorkItem(1198816, "https://devdiv.visualstudio.com/DevDiv/_workitems/edit/1198816/")]
+        public void DefiniteAssignment_UnconvertedConditionalOperator()
+        {
+            var source =
+@"class Program
+{
+    static void Main()
+    {
+        _ = new(bad) ? null : new object();
+    }
+}";
+            var comp = CreateCompilation(source);
+            comp.VerifyDiagnostics(
+                // (5,17): error CS0103: The name 'bad' does not exist in the current context
+                //         _ = new(bad) ? null : new object();
+                Diagnostic(ErrorCode.ERR_NameNotInContext, "bad").WithArguments("bad").WithLocation(5, 17)
+            );
+        }
```

Comment:
>} [](start = 8, length = 1)

We should test other expressions the rely on `BindToNaturalType()`.

For instance:
```C#
class Program
{
    static void Main()
    {
        _ = (new(bad), null) ? null : new object();
    }
}
```

And:
```C#
class Program
{
    static void F(int i)
    {
        _ = i switch { 0 => new(bad), _ => true } ? null : new object();
    }
}
```